src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs

```diff
@@ -67,12 +64,22 @@ public static int Main(string[] args)
         /// </summary>
         /// <param name="request">The Smart Detector request</param>
         /// <param name="cancellationToken">The cancellation token</param>
-        /// <returns>A <see cref="Task{TResult}"/>, returning the alerts presentations generated by the Smart Detector</returns>
-        private static async Task<List<ContractsAlert>> RunSmartDetectorAsync(SmartDetectorAnalysisRequest request, CancellationToken cancellationToken)
+        /// <returns>A <see cref="Task{TResult}"/>, returning the result of the call to the runner.</returns>
+        private static async Task<object> RunSmartDetectorAsync(SmartDetectorRunnerChildProcessInput request, CancellationToken cancellationToken)
         {
             ISmartDetectorRunner smartDetectorRunner = container.Resolve<ISmartDetectorRunner>();
             bool shouldDetectorTrace = bool.Parse(ConfigurationReader.ReadConfig("ShouldDetectorTrace", required: true));
-            return await smartDetectorRunner.RunAsync(request, shouldDetectorTrace, cancellationToken);
+
+            if (request.AnalysisRequest != null)
```

Comment:
>!= [](start = 40, length = 2)

Consider some sort of base class we 2 implementation - one for analyze and one for auto resolve, and you call the correct one from the function app.
Then you won't need to check the request type (tomorrow we can have another one, so let's use a nicer pattern than this)

Reply:
this is just a hack to be able to use the same exe for both methods

---
In reply to: [235355048](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/27#discussion_r235355048) [](ancestors = 235355048)